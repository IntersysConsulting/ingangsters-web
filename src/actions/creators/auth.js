import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGINADMIN_SUCCESS,
  LOGIN_ADMINFAIL
} from "../../actions/types/auth";
import axios from "axios";
import { API } from "../../config";

export const login = (email, password) => async dispatch => {
  const config = {
    headers: {
      "Content-Type": "application/json"
    }
  };
  const body = JSON.stringify({ email, password });
  try {
    const res = await axios.post(`${API}/user/login`, body, config);
    dispatch({
      type: LOGIN_SUCCESS,
      payload: res.data
    });
  } catch (err) {
    console.log(err);
    dispatch({
      type: LOGIN_FAIL
    });
  }
};

export const adminLogin = (email, password) => async dispatch => {
  const config = {
    headers: {
      "Content-Type": "application/json"
    }
  };
  const body = JSON.stringify({ email, password });
  try {
    const res = await axios.post(`${API}/admin/login`, body, config);
    dispatch({
      type: LOGINADMIN_SUCCESS,
      payload: res.data
    });
  } catch (err) {
    console.log(err);
    dispatch({
      type: LOGIN_ADMINFAIL
    });
  }
};
