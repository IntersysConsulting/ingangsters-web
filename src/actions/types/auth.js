export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOGIN_FAIL = "LOGIN_FAIL";
export const LOGINADMIN_SUCCESS = "LOGINADMIN_SUCCESS";
export const LOGIN_ADMINFAIL = "LOGIN_ADMINFAIL";
