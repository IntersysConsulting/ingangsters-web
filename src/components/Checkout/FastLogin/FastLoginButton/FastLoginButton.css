.FastLoginButton {
    background-color: var(--green) !important;
    border-radius: 5px !important;
    border-color: var(--green) !important;
    position: relative;
    margin: auto;
    width: 250px;
    height: 40px;
}