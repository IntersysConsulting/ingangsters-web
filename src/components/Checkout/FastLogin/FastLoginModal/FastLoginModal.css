.fastLoginModal {
    width: 270px;
    height: 170px;
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    background-color: white;
    align-content: center;
    border-radius: 5px;
}

.LoginModalContent {
    margin: auto;
}

.form-group-modal {
    height: 40px;
    margin-right: 25%;
}

.modal-body.LoginModalContent {
    padding: 0;
}

.form-control-modal {
    border-color: #000000;
    border-width: 1px;
    height: 30px;
    margin-right: 25%;
    width: 200px;
    border-radius: 5%;
}

.fastLoginModal .modal-content {
    border: 0;
}

.modal-open .modal {
    overflow-x: hidden;
    overflow-y: hidden;
}

.form-control-modal::placeholder {
    color: var(--light-grey) !important;
}

.fastLoginButtonModal {
    background-color: var(--green) !important;
    border-radius: 5px !important;
    border-color: var(--green) !important;
    width: 150px;
    height: 40px;
    display: block;
    margin: auto;
}