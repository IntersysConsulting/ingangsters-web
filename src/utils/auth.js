import axios from "axios";
import { API } from "../config";

export const isAuthenticated = () => {
  if (typeof window == "undefined") {
    return false;
  }
  return localStorage.getItem("token") ? true : false;
};

export const isAdmin = async () => {
  const token = localStorage.getItem("token");
  const config = {
    headers: {
      Authorization: `Bearer ${token}`
    }
  };
  try {
    const res = await axios.post(`${API}/admin/check`, {}, config);
    console.log("res:", res.data.isAdmin);
    return res.data.isAdmin;
  } catch (err) {
    console.log(err);
    return false;
  }
};
