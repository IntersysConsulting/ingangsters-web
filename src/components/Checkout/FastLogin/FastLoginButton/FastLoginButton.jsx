import React from "react";
import { Button } from "react-bootstrap";
import "./FastLoginButton.css";
import "../../../../css/colors.css";
<<<<<<< HEAD

const FastLoginButton = props => {
  return (
    <Button className="FastLoginButton" onClick={props.onClick}>
      Login for faster Checkout
    </Button>
  );
};

export default FastLoginButton;
=======
import { connect } from "react-redux";

const FastLoginButton = ({ isAuthenticated }) => {
  if (!isAuthenticated) {
    return (
      <Button className="FastLoginButton">Login for faster Checkout</Button>
    );
  } else {
    return <div></div>;
  }
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default connect(mapStateToProps)(FastLoginButton);
>>>>>>> 92b8b64ecb7987c77ac5ab3f81f40d0c21594e64
