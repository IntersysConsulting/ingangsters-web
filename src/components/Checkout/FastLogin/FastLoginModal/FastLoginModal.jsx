import React, { useState } from "react";
import { Button } from "react-bootstrap";
import Modal from "react-bootstrap/Modal";
import FastLoginButton from "../FastLoginButton/FastLoginButton";
import "../../../../css/colors.css";
import "./FastLoginModal.css";
import { connect } from "react-redux";
import { login } from "../../../../actions/creators/auth";

function FastLoginModal({ login, isAuthenticated }) {
  const [values, setValues] = useState({
    email: "",
    password: ""
  });
  const { email, password } = values;

  const handleChange = name => event => {
    setValues({ ...values, error: false, [name]: event.target.value });
  };

  const onSubmit = async event => {
    event.preventDefault();
    login(email, password);
  };
  const [show, setShow] = React.useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  if (!isAuthenticated) {
    console.log("Me renderizo");
    return (
      <React.Fragment>
        <FastLoginButton onClick={handleShow} />
        <Modal show={show} onHide={handleClose} className="fastLoginModal">
          <Modal.Body className="LoginModalContent">
            <form>
              <div className="form-group-modal">
                <input
                  id="emailInput"
                  onChange={handleChange("email")}
                  type="email"
                  placeholder="  Email"
                  className="form-control-modal"
                />
              </div>
              <div className="form-group-modal">
                <input
                  id="passwordInput"
                  onChange={handleChange("password")}
                  type={"password"}
                  placeholder="  Password"
                  className="form-control-modal"
                />
              </div>
            </form>
            <div className="form-button-modal">
              <Button className="fastLoginButtonModal" onClick={onSubmit}>
                Login
              </Button>
            </div>
          </Modal.Body>
        </Modal>
      </React.Fragment>
    );
  } else {
    return <div></div>;
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default connect(
  mapStateToProps,
  { login }
)(FastLoginModal);
