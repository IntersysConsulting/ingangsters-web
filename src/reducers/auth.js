import {
  LOGIN_SUCCESS,
  LOGIN_FAIL,
  LOGINADMIN_SUCCESS,
  LOGIN_ADMINFAIL,
  GET_USER,
  AUTH_ERROR
} from "../actions/types/auth";
import { isAuthenticated } from "../utils/auth";

const initialState = {
  token: localStorage.getItem("token"),
  isAuthenticated: isAuthenticated(),
  loading: true,
  user: null
};

export default function(state = initialState, action) {
  const { type, payload } = action;

  switch (type) {
    case GET_USER:
      return {
        ...state,
        isAuthenticated: true,
        loading: false,
        user: payload
      };
    case LOGIN_SUCCESS:
      localStorage.setItem("token", payload.data.token);
      return {
        ...state,
        token: localStorage.getItem("token"),
        isAuthenticated: true,
        loading: false,
        user: payload.data
      };
    case LOGINADMIN_SUCCESS:
      localStorage.setItem("token", payload.data.token);
      return {
        ...state,
        ...payload,
        token: localStorage.getItem("token"),
        isAuthenticated: true,
        loading: false
      };
    case AUTH_ERROR:
    case LOGIN_FAIL:
    case LOGIN_ADMINFAIL:
      localStorage.removeItem("token");
      return {
        ...state,
        token: null,
        isAuthenticated: false,
        loading: false,
        error: payload
      };
    default:
      return state;
  }
}
