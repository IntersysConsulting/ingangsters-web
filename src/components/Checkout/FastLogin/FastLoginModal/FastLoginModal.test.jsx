import Adapter from "enzyme-adapter-react-16";
import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import { MemoryRouter, Provider } from "react-router-dom";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import FastLoginModal from "./FastLoginModal";
import FastLoginButton from "../FastLoginButton/FastLoginButton";
import { login } from "../../../../actions/creators/auth";
import thunk from "redux-thunk";

Enzyme.configure({ adapter: new Adapter() });

describe("FastLoginModal", () => {
  const initialState = {
    auth: {
      isAuthenticated: false
    },
    show: true
  };

  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const actions = [login];
  const store = mockStore(initialState, actions);

  it("Renders Correctly", () => {
    const wrapper = renderer
      .create(<FastLoginModal store={store} dispatch={null} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("Contains two input fields and 1 Login Button", () => {
    const wrapper = mount(
      <MemoryRouter>
        <FastLoginModal store={store} dispatch={null} />
      </MemoryRouter>
    );
    wrapper.find(FastLoginButton).simulate("click");
    expect(wrapper.find("#emailInput").length).toEqual(1);
    expect(wrapper.find("#passwordInput").length).toEqual(1);
    expect(wrapper.find(FastLoginButton).length).toEqual(1);
    expect(
      wrapper.find(".form-button-modal > .fastLoginButtonModal").length
    ).toEqual(1);
  });
});
