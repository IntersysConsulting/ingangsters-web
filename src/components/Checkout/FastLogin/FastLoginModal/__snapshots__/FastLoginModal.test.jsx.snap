// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FastLoginModal Renders Correctly 1`] = `
<button
  className="FastLoginButton btn btn-primary"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Login for faster Checkout
</button>
`;
